features/lessons/src/main/java/com/cryart/sabbathschool/lessons/ui/readings/options/SSReadingDisplayOptionsView.kt

```diff
@@ -27,38 +27,46 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.databinding.DataBindingUtil
-import com.cryart.sabbathschool.core.model.SSReadingDisplayOptions
+import com.cryart.sabbathschool.core.extensions.prefs.SSPrefs
 import com.cryart.sabbathschool.lessons.R
 import com.cryart.sabbathschool.lessons.databinding.SsReadingDisplayOptionsBinding
 import com.cryart.sabbathschool.lessons.ui.base.SsBottomSheetDialogFragment
 import com.cryart.sabbathschool.lessons.ui.readings.SSReadingViewModel
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
 
+@AndroidEntryPoint
 class SSReadingDisplayOptionsView : SsBottomSheetDialogFragment() {
-    private var binding: SsReadingDisplayOptionsBinding? = null
 
-    fun setSSReadingViewModel(
-        context: Context,
-        ssReadingViewModel: SSReadingViewModel,
-        ssReadingDisplayOptions: SSReadingDisplayOptions
-    ) {
-        binding = DataBindingUtil.inflate(
-            LayoutInflater.from(context),
-            R.layout.ss_reading_display_options, null, false
-        )
-        binding?.viewModel = SSReadingDisplayOptionsViewModel(binding, ssReadingViewModel, ssReadingDisplayOptions)
+    interface ReadingDisplayOptions {
+        fun getReadingViewModel(): SSReadingViewModel
```

Comment:
You are using only one method `onSSReadingDisplayOptions` from `SSReadingViewModel`. Instead of stick to `SSReadingViewModel` try to move this method to `SSReadingDisplayOptionsViewModel`, so code will be less coupled.

Reply:
There is more that happens in `onSSReadingDisplayOptions` other than just updating prefs.
All the pages in the Viewpager need to be updated and the `SSReadingViewModel` handles this.
I will instead to for a way to observe the saved preference in `SSReadingViewModel`